File: LeetCode/14. Longest Common Prefix/Solution.java
import java.util.Arrays;

public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs == null || strs.length == 0) {
            return "";
        }
        Arrays.sort(strs);//sort the array, only compare the first and last item
        StringBuilder sb = new StringBuilder();
        int i = 0;
        int n = strs.length - 1;
        while (i < strs[0].length() && i < strs[n].length() && strs[0].charAt(i) == strs[n].charAt(i)) {
            sb.append(strs[0].charAt(i));
            i++;
        }
        return sb.toString();
    }
}

File: LeetCode/14. Longest Common Prefix/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        strs, s = sorted(strs), ''
        if not strs:
            return s

        for i, j in zip(strs[0], strs[-1]):
            if i == j:
                s += i
            else:
                break
        return s

